packages/trackerless-network/src/logic/StreamEntryPointDiscovery.ts

```diff
@@ -0,0 +1,236 @@
+        }
+    }
+
+    private keepSelfAsEntryPoint(streamPartId: string): void {
+        if (!this.config.streams.has(streamPartId) || this.servicedStreams.has(streamPartId)) {
+            return
+        }
+        this.servicedStreams.set(streamPartId, setTimeout(async () => {
+            if (!this.config.streams.has(streamPartId)) {
+                this.servicedStreams.delete(streamPartId)
+                return
+            }
+            logger.trace(`Attempting to keep self as entrypoint for ${streamPartId}`)
+            try {
+                const discovered = await this.discoverEntryPoints(streamPartId)
+                if (discovered.length < ENTRYPOINT_STORE_LIMIT 
+                    || discovered.some((peer) => isSamePeerDescriptor(peer, this.config.ownPeerDescriptor))) {
+                    await this.storeSelfAsEntryPoint(streamPartId)
+                    this.servicedStreams.delete(streamPartId)
+                    this.keepSelfAsEntryPoint(streamPartId)
+                } else {
+                    this.servicedStreams.delete(streamPartId)
+                }
+            } catch (err) {
+                logger.debug(`Failed to keep self as entrypoint for ${streamPartId}`)
+            }
+        }, this.cacheInterval))
+    }
+
+    private async avoidNetworkSplit(streamPartID: string): Promise<void> {
+        await exponentialRunOff(async () => {
+            if (this.config.streams.has(streamPartID)) {
+                const stream = this.config.streams.get(streamPartID)
+                const rediscoveredEntrypoints = await this.discoverEntryPoints(streamPartID)
+                const sortedEntrypoints = new SortedContactList(PeerID.fromString(streamPartID), 4)
+                sortedEntrypoints.addContacts(
+                    rediscoveredEntrypoints
+                        .filter((entryPoint) => !isSamePeerDescriptor(entryPoint, this.config.ownPeerDescriptor))
+                        .map((entryPoint) => new Contact(entryPoint)))
+                await Promise.allSettled(sortedEntrypoints.getAllContacts()
+                    .map((entryPoint) => stream!.layer1!.joinDht(entryPoint.getPeerDescriptor(), false)))
+                if (stream!.layer1!.getBucketSize() < 4) {
+                    throw new Error(`Network split is still possible`)
+                }
+            }
+        }, 'avoid network split', this.abortController.signal)
+        logger.trace(`Network split avoided`)
+    }
+
+    removeSelfAsEntryPoint(streamPartId: string): void {
+        if (this.servicedStreams.has(streamPartId)) {
+            setImmediate(async () => {
+                if (!this.destroyed) {
+                    await this.config.deleteEntryPointData(streamPartIdToDataKey(streamPartId))
+                }
+            })
+            clearTimeout(this.servicedStreams.get(streamPartId)!)
+            this.servicedStreams.delete(streamPartId)
+        }
+    }
+
+    async destroy(): Promise<void> {
```

Comment:
The name tells clearly that the component is not usable after the call 👍 Maybe it would make sense to rename some use this naming convention in some methods which are currently named as `stop()`?